pkgs/by-name/ba/bambu-unwrapped/package.nix

```diff
@@ -0,0 +1,242 @@
+{ runCommand
+, makeWrapper
+, buildEnv
+, fetchFromGitHub
+, gcc12Stdenv
+, lib
+, pkgsi686Linux
+, pugixml
+, libmpc
+, gmp
+, mpfr
+, boost
+, libxml2
+, xz
+, zlib
+, icu
+, suitesparse
+, mpfi
+, glpk
+, autoconf
+, autoconf-archive
+, automake
+, libtool
+, bison
+, doxygen
+, flex
+, graphviz
+, verilog
+, verilator
+, gmpxx
+, buddy
+, bash
+  # 32bit Clang with a supported major version
+, clangFrontends ? [ pkgsi686Linux.llvmPackages_16 ]
+  # 32bit GCC with a supported major version
+, gccFrontends ? [ pkgsi686Linux.gcc8 pkgsi686Linux.gcc6 pkgsi686Linux.gcc7 ]
```

Comment:
What's the advantage in having something other than the newest?

Reply:
The advantage of having multiple LLVM versions is that bambu can read LLVM IR in the textual format even if it is an older LLVM IR version. However the build currently breaks with older LLVM versions. At the start of the 2024 the build still worked with LLVM 8, 10, 11, 12 and 13 but since the last rebase on master it seems broken. As the version of bambu did not change, I suspect that something was changed in the LLVM build in nix. I was not yet able to investigate that further, but the LLVM 16 should probably suffice.

The older GCC versions can be useful, as they have different bugs and the available optimizations vary slightly. I will remove them from the defaults, as I have never needed them.